lib/utils/compose_ts.ts

```diff
@@ -775,7 +889,22 @@ async function newTarDirectory(
 				mode: fileStats.stats.mode,
 				size: fileStats.stats.size,
 			},
-			await readFile(fileStats.filePath),
+			fileStats.relPath === configPath
+				? setConfig(configPath, config, parsedBuildArgs)
+				: await readFile(fileStats.filePath),
+		);
+	}
+	if (
+		configPath === '' &&
+		(!_.isEmpty(parsedBuildArgs.global) || !_.isEmpty(parsedBuildArgs.services))
+	) {
+		const configData = setConfig('.balena/balena.yml', {}, parsedBuildArgs);
+		pack.entry(
+			{
+				name: toPosixPath('.balena/balena.yml'),
+				size: configData.length,
```

Comment:
Not 100% sure, but I think this size is meant to be in bytes, so `configData.length` would only be correct for ASCII strings. Suppose the value of a variable was UTF-8, then:
```
$ node
> '€'.length
1
> Buffer.from('€').length
3
```
Suggestion:
```
const buf = Buffer.from(configData);
pack.entry(
	{
		name: toPosixPath('.balena/balena.yml'),
		size: buf.length,
	},
	buf,
);
```
Or alternatively, dig the `pack` source code and prove to me that it's OK to have a string length as the size, then also add a comment to the source code so no else wonders if it's correct. :-)

Reply:
No, I think you are correct. An oversight on my end.